### PR TITLE
row/col stochastic matrix documentation

### PR DESCRIPTION
**Pull request by SteveBronder**
Adds docs for row and column stochastic matrix types.

I tried not to repeat myself too much by doing something similar to what Eigen does for [csr](https://eigen.tuxfamily.org/dox/group__TutorialSparse.html) matrices and csc matrices. I mostly explained the `column_stochastic_matrix` type and noted the differences for `row_stochastic_matrix` when necessary. Is that alright? 

#### Submission Checklist

- [x] Builds locally
- [ ] New functions marked with `` <<{ since VERSION }>>``
    - Is there an example of this in the docs somewhere?
- [x] Declare copyright holder and open-source license: see below

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): Simons Foundation

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC BY-ND 4.0 (https://creativecommons.org/licenses/by-nd/4.0/)

### CODE REVIEW

File: src/reference-manual/types.qmd [hunk 1]
```diff
@@ -673,6 +673,82 @@ iterations, and in either case, with less dispersed parameter
...
+$$
+\begin{bmatrix}
+0.2 & 0.5 & 0.1 & 0.2 \\
+0.2 & 0.1 & 0.6 & 0.1 \\
+0.5 & 0.2 & 0.2 & 0.1
+\end{bmatrix}
+$$
+
+
+In this example, each column(row) sums to 1, making the matrix a
```
Comment (bob-carpenter) [inline src/reference-manual/types.qmd:702]: "each row" --- this is a single example.

File: src/reference-manual/types.qmd [hunk 2]
```diff
@@ -673,6 +673,82 @@ iterations, and in either case, with less dispersed parameter
...
+0.2 & 0.5 & 0.1 & 0.2 \\
+0.2 & 0.1 & 0.6 & 0.1 \\
+0.5 & 0.2 & 0.2 & 0.1
+\end{bmatrix}
+$$
+
+
+In this example, each column(row) sums to 1, making the matrix a 
+valid `column_stochastic_matrix` and `row_stochastic_matrix`.
+
```
Comment (bob-carpenter) [inline src/reference-manual/types.qmd:704]: The example you gave is only a valid row-stochastic matrix.  I'd just use the words here rather than quoted types, too.

File: src/reference-manual/types.qmd [hunk 3]
```diff
@@ -673,6 +673,82 @@ iterations, and in either case, with less dispersed parameter
...
+Column stochastic matrices are often used in models where 
+each column represents a probability distribution across a 
+set of categories, such as in multiple multinomial distributions, 
+transition matrices in Markov models, or compositional data analysis. 
+They can also be used in situations where multiple Dirichlet-distributed v
+ariables are required across different dimensions.
```
Comment (bob-carpenter) [inline src/reference-manual/types.qmd:710]: premature line break

File: src/reference-manual/types.qmd [hunk 4]
```diff
@@ -673,6 +673,82 @@ iterations, and in either case, with less dispersed parameter
...
+
+
+In this example, each column(row) sums to 1, making the matrix a 
+valid `column_stochastic_matrix` and `row_stochastic_matrix`.
+
+Column stochastic matrices are often used in models where 
+each column represents a probability distribution across a 
+set of categories, such as in multiple multinomial distributions, 
+transition matrices in Markov models, or compositional data analysis. 
+They can also be used in situations where multiple Dirichlet-distributed v
```
Comment (bob-carpenter) [inline src/reference-manual/types.qmd:709]: I'd drop the Dirichlet comment here.  You can just say they can be used whenever you need multiple simplexes of the same dimensionality.  

The other big application here is factor models, so you should definitely mention those.  The rows in the row stochastic matrix in these models is, for exmaple, is something like the proportion of pollutants being emitted from a factory.

File: src/reference-manual/types.qmd [hunk 5]
```diff
@@ -673,6 +673,82 @@ iterations, and in either case, with less dispersed parameter
...
+ariables are required across different dimensions.
+
+The `column_stochastic_matrix` and `row_stochastic_matrix` types are declared 
+with full dimensionality. For instance, a matrix `theta` with
```
Comment (bob-carpenter) [inline src/reference-manual/types.qmd:713]: full dimensionality ---> row and column sizes

File: src/reference-manual/types.qmd [hunk 6]
```diff
@@ -673,6 +673,82 @@ iterations, and in either case, with less dispersed parameter
...
+The `column_stochastic_matrix` and `row_stochastic_matrix` types are declared 
+with full dimensionality. For instance, a matrix `theta` with 
+3 rows and 4 columns, where each 
+column is a 3-simplex, is declared as:
+
+```stan
+column_stochastic_matrix[3, 4] theta;
+```
+
+A matrix `theta` with
```
Comment (bob-carpenter) [inline src/reference-manual/types.qmd:721]: Too many line breaks.

File: src/reference-manual/types.qmd [hunk 7]
```diff
@@ -673,6 +673,82 @@ iterations, and in either case, with less dispersed parameter
...
+
+A matrix `theta` with 
+3 rows and 4 columns, where each 
+row is a 4-simplex, is declared as:
+
+```stan
+row_stochastic_matrix[3, 4] theta;
+```
+
+As with simplexes, `column(row)_stochastic_matrix` variables are subject to
```
Comment (bob-carpenter) [inline src/reference-manual/types.qmd:729]: This is too hard to parse---just repeat, we have tons of space here.  That is, separate out column_stochastic_matrix and row_stochastic_matrix.

File: src/reference-manual/types.qmd [hunk 8]
```diff
@@ -673,6 +673,82 @@ iterations, and in either case, with less dispersed parameter
...
+```
+
+As with simplexes, `column(row)_stochastic_matrix` variables are subject to 
+validation, ensuring that each column(row) satisfies the simplex constraints. 
+This validation accounts for floating-point imprecision, with checks 
+performed up to a statically specified accuracy threshold \(\epsilon\).
+
+#### Stability Considerations {-}
+
+In high-dimensional settings or when the matrix has many columns,
```
Comment (bob-carpenter) [inline src/reference-manual/types.qmd:736]: It's not clear how "high-dimensional" and "has many columns" differ.  Is high-dimensional the rows?  I'd just say "high-dimensional" here.

**Review by bob-carpenter — CHANGES_REQUESTED** [review]
Thanks for writing this up.  I had a bajillion little detailed comments here---sorry about that in advance.

File: src/reference-manual/types.qmd (hunk 6 above)
Comment (SteveBronder) [reply src/reference-manual/types.qmd]: Fixed

File: src/reference-manual/types.qmd (hunk 5 above)
Comment (SteveBronder) [reply src/reference-manual/types.qmd]: Fixed

File: src/reference-manual/types.qmd (hunk 3 above)
Comment (SteveBronder) [reply src/reference-manual/types.qmd]: Cut sentence

File: src/reference-manual/types.qmd (hunk 4 above)
Comment (SteveBronder) [reply src/reference-manual/types.qmd]: Added factor model to the examples

File: src/reference-manual/types.qmd (hunk 2 above)
Comment (SteveBronder) [reply src/reference-manual/types.qmd:704]: I tried to reword this a bit to show I am talking about both examples. Should I just break this up into two sentences?

File: src/reference-manual/types.qmd (hunk 1 above)
Comment (SteveBronder) [reply src/reference-manual/types.qmd]: I fixed this a bit to make it more clear we are talking about both of the examples

File: src/reference-manual/types.qmd [hunk 9]
```diff
@@ -673,6 +673,82 @@ iterations, and in either case, with less dispersed parameter
 initialization or custom initialization if there are informative
 priors for some parameters.
 
+### Stochastic Matrices {-}
+
+A stochastic matrix is a matrix where each column, row, or both is a 
+unit simplex, meaning that each column(row) vector has non-negative 
+values that sum to 1. For example, a \(3 \times 4\) 
+column stochastic matrix will look like:
```
[the remark this answers is not in the document]
Comment (SteveBronder) [reply src/reference-manual/types.qmd]: Also fixed Column-matrix below

File: src/reference-manual/transforms.qmd [hunk 10]
```diff
@@ -672,6 +672,72 @@ z_k
...
+X_{ij} \geq 0 \quad \text{for } 1 \leq i \leq N,
+$$
+
+and
+
+$$
+\sum_{i=1}^N X_{ij} = 1.
+$$
+
+A row stochastic matrix is defined similarly but with the axis flipped such
```
[the remark this answers is not in the document]
Comment (SteveBronder) [reply src/reference-manual/transforms.qmd]: Agree, can't believe I forgot to just say what this is in simple terms

File: src/reference-manual/transforms.qmd [hunk 11]
```diff
@@ -672,6 +672,72 @@ z_k
 .
 $$
 
+## Stochastic Matrix {#stochastic-matrix-transform.section}
+
+The `column_stochastic_matrix[N, M]` and `row_stochastic_matrix[N, M]` type in
```
[the remark this answers is not in the document]
Comment (SteveBronder) [reply src/reference-manual/transforms.qmd]: discussed irl, we are wishy washy with col vs column abbreviation, but I like column here so going to keep it

**Comment by WardBrian** [comment]
The discussion in the HMM thread reminded me: do we think this will be ready by Tuesday for when we build the docs for the new release?

**Comment by SteveBronder** [comment]
Updating this now
